features/mbedtls/targets/TARGET_STM/aes_alt.c

```diff
@@ -182,21 +247,41 @@ int mbedtls_aes_crypt_cbc( mbedtls_aes_context *ctx,
     int status = 0;
     if( length % 16 )
         return( MBEDTLS_ERR_AES_INVALID_INPUT_LENGTH );
+    ctx->hcryp_aes.Init.pInitVect = &iv[0];
+    status |= st_cbc_restore_context(ctx);
 #if defined (TARGET_STM32L486xG)
+    uint32_t *iv_ptr = (uint32_t *)&iv[0];
     if( mode == MBEDTLS_AES_DECRYPT ) {
-        status = st_hal_cryp_cbc(ctx, CRYP_ALGOMODE_KEYDERIVATION_DECRYPT, length, iv, (uint8_t *)input, (uint8_t *)output);
+        status |= st_hal_cryp_cbc(ctx, CRYP_ALGOMODE_KEYDERIVATION_DECRYPT, length, iv, (uint8_t *)input, (uint8_t *)output);
```

Comment:
I see that the `input` parameter is marked as `const`. Is there a risk that this function will actually modify the input buffer contents? If this is the case, this will probably change the behaviour of the mbed TLS function